### PR TITLE
Update gradle and add jfx plugin

### LINKED ISSUE #636: shemnon javafx plugin no longer supported
I will look into an alternative.

https://bitbucket.org/shemnon/javafx-gradle/issues/47/adding-manifest-attribute-javafx#comment-24360784

### PR DESCRIPTION
**Pull request by AustinShalit**
closes #636 

Updates gradle to 2.14.1 and also setup the jfx plugin to replace the shemnon javafx plugin

### CODE REVIEW

**Comment by AustinShalit** [comment]
Tested the task :ui:jfxNative on both Mac and Windows.  Works as expected

**Comment by AustinShalit** [comment]
GTG

File: .travis.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ deploy:
   api_key:
     secure: bdjiFXG3VBcM1iyxeputsWA9vHxgLNcxRB1i0REDNAnIEvDWpAdVr1jPBejeiOEw9s+k6EItATvQ2I7Xp5iKnYaXn0NuHzM3OSY8WNVCc+gBTaQwkijPBqi/vEXaO502CscW+H2U26QZGS/jJiUuieeErtzCNcWmCv8SRc5NFOVYdSlhAmp+aPeu2YwuweiMZNVquYM2hx7murghOYSF5hQTKq4/zrziENVi78XZS8rJFmxwOGqNj1GBLul2F9poIPjzDbPVJLeXTJEY+N3aYGJ4GyZYf6p8ynDO/v043/GuOAyNfkgSwhCcTQlmLG/mm6P9CUMoDQR9klh2eDswgrIP4rGLwILg3035nqHqwgTIKVXfspni3P/65siS6lXV0M5flEkRysMBr1GQhaIEUM32ArreMzAtC2Vn2CviKG0TwLmxyd/7W3JdoX4KrHAcKBwbqFDImWw43S6N63oTUEw6GjrI+CndwgF9RNgr5SAyS28LFXViOJe7JlGVUNU33kMNpEJxzVREzkJsO8aicaZTrvDniFOfg2WhrAyhjxmh4/IOHDE6qgtxw+2+TUzMlwdzdxb90BeXg2XfMichmKfQYQFEmZntCZcoHHDG5jbCTyAvFuOv6tdPjpr69QjiEJhEbohw8CEwoGhy1kxxQcCDCfABrOTivJ39Ui8MC6k=
   file_glob: true
-  file: "ui/build/distributions/*.{deb,pkg,dmg}"
+  file: "ui/build/jfx/native/*.{deb,pkg,dmg}"
```
Comment (JLLeitschuh) [inline .travis.yml]: Thank you! I would have forgotten this!

**Comment by JLLeitschuh** [comment]
Has this been tested on a windows system?

**Comment by SamCarlberg** [comment]
@JLLeitschuh [Looks like it](https://github.com/WPIRoboticsProjects/GRIP/pull/637#issuecomment-234767005)

**Comment by AustinShalit** [comment]
- [*] Mac
- [  ] Windows x32
- [*] Windows x64
- [  ] Linux

**Comment by AustinShalit** [comment]
@ThomasJClark,
If you have a few minutes to lend a hand, I am having trouble getting GRIP to assemble on Ubuntu.  Here are the steps I am taking right now to try to get GRIP to assemble:

1. I built [ntcore](https://github.com/wpilibsuite/ntcore) locally using the parameter makeDesktop (`./gradlew publish -PmakeDesktop`)
2. I copied the NetworkTables-3.0.0-SNAPSHOT-desktop.jar artifact into core/libs and modified my build.gradle to match the one in #402 
3. I assemble grip (`./gradlew jfxNative`)

When I run the GRIP executable, I get this message in my console:
```
terminate called after throwing an instance of 'std::system_error'
  what():  Enable multithreading to use std::thread: Operation not permitted
```

When I run the GRIP jar that is put in the executable GRIP runs as expected.

If you could give me some advice on what I am doing wrong I would really appreciate it!
Thank you,
Austin

**Comment by SamCarlberg** [comment]
This is why I hate that network tables doesn't have a pure Java version

**Comment by JLLeitschuh** [comment]
@AustinShalit This is expected. @ThomasJClark had a version of Network Tables that he built himself. This needs to be fixed before our next release probably.
This is fine for now though.

**Comment by AustinShalit** [comment]
@PeterJohnson do you have any thoughts on how to get this to work?

**Comment by PeterJohnson** [comment]
We know this error is due to a missing -pthread flag somewhere when either compiling or linking but it's not clear where it's being left out since we updated the ntcore build to supposedly put it in.  I have an Ubuntu system so I will give this a try tonight to see if I can reproduce it.

**Comment by AustinShalit** [comment]
Ok. Thank you!

**Comment by PeterJohnson** [comment]
The underlying issue is that the GRIP executable is simply a copy of a standard one that doesn't link to libpthread.  While LD_PRELOAD can be used to work around the issue, the right solution is to compile the GRIP executable with -pthread from this source code: http://hg.openjdk.java.net/openjfx/8u-dev/rt/file/fe6b492b9150/modules/fxpackager/src/main/native/launcher/linux/launcher.cpp
(e.g. `g++ -o GRIP -pthread launcher.cpp -ldl`)

**Pull request merged**